wicket-core/src/main/java/org/apache/wicket/page/PartialPageUpdate.java

```diff
@@ -567,7 +567,7 @@ public final void add(final Component component, final String markupId)
 				LOG.warn("Component '{}' not cannot be updated because it was already removed from page", component);
 				return;
 			}
-			else if (pageOfComponent != page) 
+			else if (!pageOfComponent.equals(page))
```

Comment:
Do you have a usecase, why this change should be necessary?

The check is precisely there to check on the page instance.

Reply:
Sorry, I have missed the most important part in the pull request.
So now I have added another commit containing the `equals()` and `hashCode()` methods for the Page.class.

Well my usecase is the following:
I have a `AbstractAjaxTimerBehavior` on my page, which runs every 500 milli secs.
This `AbstractAjaxTimerBehavior` pulls Buttons out of a registry.
The registry contains the Buttons together with an information, whether the button should be enabled or disabled.
The state of the button enabled/disabled can be updated via a websocket, that is connected to the server.
So when the `AbstractAjaxTimerBehavior` pulls the Buttons our of the registry, the buttons are assigned to the same page, but it is not the very same object in the heap. 
Thus the inequality check with `pageOfComponent != page` goes true, even though the page is the same.
This is why I would like to have the inequality check as `!pageOfComponent.equals(page)` with the `equals()` method checking on the immutal fields of the page.

Do you agree, that this is a sensible approach, or would you suggest otherwise?

Reply:
Changing #equals and #hashCode in such a central class can have severe negative consequence, that I would not like to risk.
Your idea with updating via component path sounds good to me.

Reply:
ok, understood.
Anyhow, thanks for the support and quick responses! 👍